antlr-kotlin-target/src/main/resources/org/antlr/v4/tool/templates/codegen/Kotlin/Kotlin.stg

```diff
@@ -801,11 +806,11 @@ StructDecl(struct, ctorAttrs, attrs, getters, dispatchMethods, interfaces, exten
 public open class <struct.name> : <if(contextSuperClass)><contextSuperClass><else>ParserRuleContext<endif><if(interfaces)> : <interfaces; separator=", "><endif> {
     override val ruleIndex: Int = Rules.<struct.derivedFromName; format="cap">
 
-    <attrs:{a | public var <a>}; separator="\n">
+    <attrs:{a | @JvmField public var <a>}; separator="\n">
```

Comment:
This change may break existing code, which I think is acceptable, but I just wonder if it could make sense to first replicate the tests we have in ANTLR 5, so that we are more confident this change will work on all platforms before introducing it

Reply:
Also, this avoid the platform clash on the JVM, but does it work also on other platforms?

Reply:
> This change may break existing code

I think that's not something you have to take into account, as I really doubt a Java consumer is using this Kotlin target.

>  so that we are more confident this change will work on all platforms before introducing it

I have tested with code specifically written to reproduce conflicting signatures, and all supported platforms compiled and run correctly.

> Also, this avoid the platform clash on the JVM, but does it work also on other platforms?

The problem is specific to the JVM. In JS/TS you end up with code similar to:

```
class Example {
    get foo(): number;
    getFoo(): string;
}
```

Reply:
Ok, good

Reply:
@piacenti ahh no, this is what I did test, your case is a new one.